File: tooling/intellij/stratos-language/src/main/kotlin/com/araizen/stratos/files/StratosFileTypeFactory.java
package com.araizen.stratos.files;

import com.intellij.openapi.fileTypes.FileTypeConsumer;
import com.intellij.openapi.fileTypes.FileTypeFactory;
import org.jetbrains.annotations.NotNull;

/**
 * Note: This class is only used with the fileTypeFactory extension point
 * for versions of the IntelliJ Platform prior to v2019.2
 */
public class StratosFileTypeFactory extends FileTypeFactory {
    @Override
    public void createFileTypes(@NotNull FileTypeConsumer fileTypeConsumer) {
        fileTypeConsumer.consume(StratosFileType.INSTANCE);
    }
}

File: tooling/intellij/old/src/main/kotlin/com/araizen/stratos/files/StratosFileTypeFactory.kt
package com.araizen.stratos.files

import com.araizen.stratos.files.StratosFileType
import com.intellij.openapi.fileTypes.FileTypeConsumer
import com.intellij.openapi.fileTypes.FileTypeFactory

/**
 * Note: This class is only used with the fileTypeFactory extension point
 * for versions of the IntelliJ Platform prior to v2019.2
 */
class StratosFileTypeFactory : FileTypeFactory() {
    override fun createFileTypes(fileTypeConsumer: FileTypeConsumer) {
        fileTypeConsumer.consume(StratosFileType.INSTANCE)
    }
}